### PR TITLE
sundials: add v6.4.1, new ginkgo and kokkos variants, plus some fixes

### PR DESCRIPTION
**Pull request by balos1**
cc @balay

### CODE REVIEW

File: var/spack/repos/builtin/packages/sundials/package.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     # Versions
     # ==========================================================================
     version("develop", branch="develop")
+    version("6.4.1", sha256="7bf10a8d2920591af3fba2db92548e91ad60eb7241ab23350a9b1bc51e05e8d0")
```
Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py:30]: Add 6.4.0 as well for completeness

File: var/spack/repos/builtin/packages/sundials/package.py [hunk 2]
```diff
@@ -122,9 +123,10 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
         when="@6.0.0: +profiling",
         description="Enable Caliper instrumentation/profiling",
     )
+    variant("ginkgo", default=False, when="@6.4.0:", description="Enable Ginkgo interfaces")
     variant("hypre", default=False, when="@2.7.0:", description="Enable Hypre MPI parallel vector")
-    variant("lapack", default=False, description="Enable LAPACK direct solvers")
     variant("klu", default=False, description="Enable KLU sparse, direct solver")
+    variant("lapack", default=False, description="Enable LAPACK direct solvers")
```
Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py:137]: Add Kokkos option as well?

**Review by gardner48 — CHANGES_REQUESTED** [review]
Minor requests to add 6.4.0 and a Kokkos variant, otherwise LGTM

**Review by gardner48 — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Confirmed 2 new sha256.

**Pull request merged**